hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSecondaryIndexPruning.scala

```diff
@@ -105,7 +138,210 @@ class TestSecondaryIndexPruning extends SecondaryIndexTestBase {
     }
   }
 
-  private def checkAnswer(sql: String)(expects: Seq[Any]*): Unit = {
-    assertResult(expects.map(row => Row(row: _*)).toArray.sortBy(_.toString()))(spark.sql(sql).collect().sortBy(_.toString()))
+  @Test
+  def testSecondaryIndexPruningWithUpdates(): Unit = {
+    if (HoodieSparkUtils.gteqSpark3_3) {
+      var hudiOpts = commonOpts
+      hudiOpts = hudiOpts + (
+        DataSourceWriteOptions.TABLE_TYPE.key -> HoodieTableType.COPY_ON_WRITE.name(),
```

Comment:
Also add a test with a permutation of MOR table?

Reply:
lets also add a test for non partitioned table